src/sentry/snuba/profile_functions_metrics.py

```diff
@@ -0,0 +1,298 @@
+):
+    builder = ProfileFunctionsMetricsQueryBuilder(
+        dataset=Dataset.PerformanceMetrics,
+        params=params,
+        snuba_params=snuba_params,
+        query=query,
+        selected_columns=selected_columns,
+        equations=equations,
+        orderby=orderby,
+        limit=limit,
+        offset=offset,
+        sample_rate=sample,
+        config=QueryBuilderConfig(
+            auto_fields=auto_fields,
+            auto_aggregations=auto_aggregations,
+            use_aggregate_conditions=use_aggregate_conditions,
+            functions_acl=functions_acl,
+            equation_config={"auto_add": include_equation_fields},
+            has_metrics=has_metrics,
+            use_metrics_layer=use_metrics_layer,
+            transform_alias_to_input_format=transform_alias_to_input_format,
+            skip_tag_resolution=skip_tag_resolution,
+        ),
+    )
+
+    result = builder.process_results(builder.run_query(referrer))
+    return result
+
+
+def timeseries_query(
+    selected_columns: list[str],
+    query: str,
+    params: ParamsType,
+    rollup: int,
+    referrer: str,
+    snuba_params: SnubaParams | None = None,
+    zerofill_results: bool = True,
+    allow_metric_aggregates=True,
+    comparison_delta: timedelta | None = None,
+    functions_acl: list[str] | None = None,
+    has_metrics: bool = True,
+    use_metrics_layer: bool = False,
+    on_demand_metrics_enabled: bool = False,
+    on_demand_metrics_type: MetricSpecType | None = None,
+    groupby: Column | None = None,
+) -> SnubaTSResult:
+    """
+    High-level API for doing arbitrary user timeseries queries against events.
+    this API should match that of sentry.snuba.discover.timeseries_query
+    """
+
+    if len(params) == 0 and snuba_params is not None:
+        params = snuba_params.filter_params
+
+    metrics_query = TimeseriesProfileFunctionsMetricsQueryBuilder(
+        params,
+        rollup,
+        snuba_params=snuba_params,
+        dataset=Dataset.PerformanceMetrics,
+        query=query,
+        selected_columns=selected_columns,
+        groupby=groupby,
+        config=QueryBuilderConfig(
+            functions_acl=functions_acl,
+            allow_metric_aggregates=allow_metric_aggregates,
+            use_metrics_layer=use_metrics_layer,
+        ),
+    )
+    result = metrics_query.run_query(referrer)
+
+    result = metrics_query.process_results(result)
+    result["data"] = (
+        discover.zerofill(
+            result["data"],
+            params["start"],
+            params["end"],
+            rollup,
+            ["time"],
+        )
+        if zerofill_results
+        else result["data"]
+    )
+
+    result["meta"]["isMetricsData"] = True
+
+    return SnubaTSResult(
+        {
+            "data": result["data"],
+            "isMetricsData": True,
+            "meta": result["meta"],
+        },
+        params["start"],
+        params["end"],
+        rollup,
+    )
+
+
+def top_events_timeseries(
```

Comment:
just a heads up, if you plan to use top events for this using the events_stats endpoint, you'll need to add your dataset to this list https://github.com/getsentry/sentry/blob/master/src/sentry/api/endpoints/organization_events_stats.py#L247-L256